fission/function.go

```diff
@@ -66,6 +65,47 @@ func createArchive(client *client.Client, fileName string) *fission.Archive {
 	return &archive
 }
 
+func createPackage(client *client.Client, envName, srcArchiveName, deployArchiveName, buildcmd string) *api.ObjectMeta {
+	pkgSpec := fission.PackageSpec{
+		Environment: fission.EnvironmentReference{
+			Namespace: api.NamespaceDefault,
+			Name:      envName,
+		},
+	}
+	var pkgStatus fission.BuildStatus = fission.BuildStatusSucceeded
+
+	if len(deployArchiveName) > 0 {
+		pkgSpec.Deployment = *createArchive(client, deployArchiveName)
+		if len(srcArchiveName) > 0 {
+			fmt.Println("Deployment may be overwritten by builder manager after source package compilation")
+		}
+	}
+	if len(srcArchiveName) > 0 {
+		pkgSpec.Source = *createArchive(client, srcArchiveName)
+		// set pending status to package
+		pkgStatus = fission.BuildStatusPending
+	}
+
+	if len(buildcmd) > 0 {
+		pkgSpec.BuildCommand = buildcmd
+	}
+
+	pkgName := strings.ToLower(uuid.NewV4().String())
```

Comment:
UUIDs are hard to work with on the CLI. Why not accept a `name` parameter? And for implicitly created packages (from functions), use `<functionName>-<uniuri.NewLen(6)>`

Reply:
I see what you're saying. How about we do this:
 1. Packages created implicitly during function create/update take the function name + unique id
 1. Users are allowed to rename packages with the upcoming package cli
 1. Users are allowed to name packages during creating if they use the package cli 
 1. Package metadata is annotated with a list of functions that use it
 1. Package metadata is labelled with a "createdForFunction=<function name>" label when it's created implictly by a function create